### PR TITLE
Add support to SSL/TLS options

### LINKED ISSUE #44: Add SSL suport
Please post all product and debugging questions on our [forum](https://discuss.elastic.co/c/logstash). Your questions will reach our wider community members there, and if we confirm that there is a bug, then we can open a new issue here.

For all general issues, please provide the following details for fast resolution:

- Version:
- Operating System:
- Config File (if you have sensitive info, please remove it):
- Sample Data:
- Steps to Reproduce:

### LINKED ISSUE #14933: Logstash - Plugins - SSL Settings Standardization - Standardize Redis OUTPUT settings
This issue should make the plugin SSL settings consistent with the naming convention defined in the [meta issue](https://github.com/elastic/logstash/issues/14905). Missing settings already supported by the plugin's HTTP/TCP client, which only requires small coding changes on the plugin source, should also be added through this issue.

**Add and deprecate options to comply with the naming convention**
- ssl -> ssl_enabled

**Add common settings validations**
- Not allow mixing the same deprecated and new setting on the same plugin configuration, e.g. `ssl` and `ssl_enabled`

**Add missing settings/functionality that fits into the phase 1 description, and move non-phase 1-2 to the phase 3 issue (to be created if doesn't exist)**
- ssl_certificate
- ssl_certificate_authorities
- ssl_cipher_suites
- ssl_handshake_timeout
- ssl_key
- ssl_key_passphrase
- ssl_supported_protocols
- ssl_verification_mode
- ssl_keystore_path
- ssl_keystore_password
- ssl_keystore_type
- ssl_keystore_key_password
- ssl_truststore_path
- ssl_truststore_password
- ssl_truststore_type

### PR DESCRIPTION
**Pull request by edmocosta**
- Added support to SSL/TLS configurations (`ssl_enabled`, `ssl_certificate_authorities`, `ssl_certificate`, `ssl_key`, `ssl_verification_mode`, `ssl_supported_protocols` and `ssl_cipher_suites`)
 - Fixed integration tests

---
Closes: https://github.com/elastic/logstash/issues/14933
Closes: https://github.com/logstash-plugins/logstash-output-redis/issues/44

### CODE REVIEW

**Comment by jelmokri** [comment]
Can you merge it and release please ?

File: lib/logstash/outputs/redis.rb [hunk 1]
```diff
@@ -199,6 +229,67 @@ def connect
...
+        # if we have an encrypted key and a password is not provided (nil) than OpenSSL::PKey::RSA
+        # prompts the user to enter a password interactively - we do not want to do that,
+        # for plain-text keys the default '' password argument gets simply ignored
+        params[:key] = OpenSSL::PKey::RSA.new(File.read(@ssl_key), @ssl_key_passphrase.value || '')
+      end
+    end
+
+    params[:min_version] = :TLS1_1
+    if @ssl_supported_protocols.any?
+      protocols = @ssl_supported_protocols.map { |v| "TLS#{v[-3..-1].gsub('.', '_')}".to_sym }.sort
```
Comment (jsvd) [inline lib/logstash/outputs/redis.rb]: We can make the intent here a bit easier to read:
```suggestion
      protocols = @ssl_supported_protocols.map { |v| v.delete('v').tr(".", "_").to_sym }.sort
```

**Review by jsvd — APPROVED** [review]
Minor nitpick on idiomatic ruby.
I've tested locally with multiple TLS setups including plain cert+key and root ca, using an intermediate certificate, both with separate root + intermediate certificate files or a single certificate bundle.
Everything seems to work correctly. LGTM

**Comment by edmocosta** [comment]
Hi @jelmokri!

I've just published a new version ([5.2.0)](https://rubygems.org/gems/logstash-output-redis/versions/5.2.0) that includes this change:

```
bin/logstash-plugin install --version 5.2.0 logstash-output-redis
```

Thanks!

**Comment by jelmokri** [comment]
Is it possible to release it in new logstash version ? 
I tested on 8.14.1 and ssl parameter still not available

**Comment by edmocosta** [comment]
Hi @jelmokri!

Logstash 8.14.1 is still embedding the previous version of this plugin (`5.0.0`). It will be bumped to the latest in the upcoming releases. 

For now, you can update the plugin manually by running:

```
bin/logstash-plugin update logstash-output-redis
```

**Pull request merged**